### DISCUSSION TITLE
Terraform provider updates won't automerge

### DISCUSSION THREAD

**Discussion by TheKevJames** [renovatebot/renovate#20594] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I'm attempting to enable automerge for various terraform providers.

### What you tried so far.

I'm trying to track down why [this PR](https://github.com/TheKevJames/infrastructure/pull/202) didn't get configured for automerging. That repo's renovate config includes `"extends": ["github>thekevjames/tools:personal"]`, which refers to [this config](https://github.com/TheKevJames/tools/blob/master/personal.json) which should be applying [this package rule](https://github.com/TheKevJames/tools/blob/22d864a129d45bb1ba40158db1372edb91190543/trustedpackages.json#L4-L14). However, I can't seem to come up with the correct invocation to have terraform's `required_provider` automerge via this spec.

### Relevant debug logs

[Full debug logs](https://gist.github.com/TheKevJames/1b716cfbcd4732cd0b0b60f80ee598fa) from the relevant job.

In case Renovate staff with access get involved in this, the relevant dashboard link is [here](https://app.renovatebot.com/dashboard#github/TheKevJames/infrastructure/1007676989) with Job ID 1007676989.

**Comment by rarkins** [comment, marked as answer]
In this case the package is aliased as `google`, so you should match on that string instead of `hashicorp/google`.

**Comment by TheKevJames** [comment]
I've updated my config accordingly ([here](https://github.com/TheKevJames/tools/commit/15885421f672e1ab72e78cb825836f1e90248bca)) and will mark as accepted once I verify that works.

@rarkins so does that mean that the `packageNames` value in a renovate config is actually looking at the `depName` instead of the `packageName`? ie. given the renovate logs showing:
```
{
  "datasource": "terraform-provider",
  "depIndex": 1,
  "depName": "google",
  "depType": "provider",
  "packageName": "hashicorp/google",
  "skipReason": "no-version",
  "updates": []
}
```

I attempted to look this up in [the docs](https://docs.renovatebot.com/configuration-options/#packagerules) but it seems like the `packageNames` config value is not documented?

**Reply by rarkins** [reply]
You should be using `matchPackageNames`, not `packageNames`. And yes currently they match against `depName` but we're trying to phase that out.

**Reply by TheKevJames** [reply]
Ah, gotcha, I guess the field was renamed at some point. Sounds good, thanks!

**Comment by TheKevJames** [comment]
Noting back that this is indeed now working, though I'm still a bit confused about the docs mismatch. Might be something worth cleaning up in the docs to avoid future confusion!

**Reply by viceice** [reply]
feel free to update docs, they are all here inside this repo 🤗

**Reply by TheKevJames** [reply]
If I had a clear understanding of this, I'd be more than happy to :P

rarkins said in the thread above:
> And yes currently they match against `depName` but we're trying to phase that out.

So if its WIP to fix this anyway, maybe not worth the update.
